hal/network/ncp_client/sara/sara_ncp_client.cpp

```diff
@@ -1076,9 +1076,16 @@ int SaraNcpClient::selectSimCard(ModemState& state) {
                 }
 
                 // This is a persistent setting
-                int umnoprof = static_cast<int>(UbloxSaraUmnoprof::SIM_SELECT);
+                umnoprof = static_cast<int>(UbloxSaraUmnoprof::SIM_SELECT);
                 if (netConf_.netProv() == CellularNetworkProvider::TWILIO) { // if Twilio Super SIM
                     umnoprof = static_cast<int>(UbloxSaraUmnoprof::STANDARD_EUROPE);
+                } else {
+                    // break out of do-while if we're trying to set SIM_SELECT a second time
+                    if (resetCount >= 1) {
```

Comment:
Based on the PR description, shouldn't this workaround only work when the external SIM is used? Or is it not that important to have a correct profile selected for officially supported providers other than Twilio?

Reply:
Currently the only other SIM we only support on R410 is Kore AT&T, which does get picked up via the SIM_SELECT as AT&T profile.  SIM_SELECT is used as a way to somewhat provide 3rd party SIM support if it works... if not... then it will default now to SW_DEFAULT 0 profile.

Reply:
In other words, it's ok if the client is unable to set SIM_SELECT for Kore on the first try, correct?

Reply:
Should we maybe make it try at least twice?

Reply:
I would stick to the previous behavior as much as possible while still addressing the very specific issue the customer is having. If I'm not mistaken 2.0.0 is the first release that introduces this reset-and-retry loop when setting the operator profile, and, most importantly, it also fails the connection attempt if it wasn't able to set the profile after a number of attempts, which is exactly what is causing the issue.

I'm assuming that there's a good reason why we introduced this loop, so what if after making a few attempts to set the profile we simply ignore the fact that the profile wasn't set correctly and proceed with the initialization just like previous Device OS versions did?

Reply:
Yes that's what it's been changed to do now, and to be more like Gen 2 was previously (before 2.0.0) by trying a few times in case of certain errors.  But really this retry loop is in case we set UMNOPROF=x and it keeps ending up in SW_DEFAULT. I'm seeing a couple other cases that need to be resolved automatically as well that I will fix in the next push. Regarding this thread though, I think I will bump the `if (resetCount >= 2) {` to put more emphasis on doing the right thing ™️  for official SIMs, and less emphasis on speeding unofficial SIMs to default to SW_DEFAULT.

Reply:
The current code is more like
- using Twilio SIM: fail the initialization if unable to set the profile
- using any other SIM including Kore: try setting the profile but ignore the error if it fails

Is there a reason to treat Twilio and Kore differently?

Reply:
That's the "bug" part of this fix :D

Reply:
I would be fine with it, since I'm almost certainly missing some context here :sweat_smile: I just wanted to make sure this is exactly the behavior we wanted

Reply:
I'm going to merge this as it worked mostly ok as-is, but with the other UMNOPROF=100 PR and these changes together I'm noticing some more case that need to be addressed.  So we can pick up the review over there :D

Reply:
Sure. I have an ask: if this is going to land in 2.0.0 as it is implemented now, let's add a comment that would state that it's an intended behavior, as that's not obvious from the code (hence why it caught my eye).

Reply:
It will change, but I have two PR's open for similar parts of the code which is making life difficult.  Since that other one has more changes... I'll merge this one, rebase that one and make the changes.  I'll try to make it very obvious what the intended behavior is with comments.

Reply:
Cool. Thank you :pray: